File: 2022/day1.py
from aocd import data
from pathlib import Path

txtfile =  Path(__file__).parent / (Path(__file__).stem + ".txt")
if not txtfile.exists():
    with open(txtfile, 'w') as f:
        f.write(data)

with open(txtfile) as f:
    data = [sum([int(snack) for snack in elf.split("\n")]) for elf in f.read().split("\n\n")]
elfs = sorted(data, reverse=True)

print(f"Part 1: {elfs[0]}")
print(f"Part 2: {sum(elfs[0:3])}")

File: 2022/rust/src/day1.rs
const INPUT: &str = include_str!("day1.txt");

fn input() -> Vec<Vec<u32>> {
    INPUT
        .split("\r\n\r\n")
        .map(|line| {
            line.split("\r\n")
                .map(|value| value.parse().unwrap())
                .collect::<Vec<u32>>()
        })
        .collect::<Vec<Vec<u32>>>()
}

#[test]
fn solve() {
    let mut elfs = input()
        .iter()
        .map(|elf| elf.iter().sum())
        .collect::<Vec<u32>>();
    elfs.sort();
    elfs.reverse();
    println!("Part 1: {:?}", elfs.first().unwrap());
    println!("Part 1: {:?}", elfs[0] + elfs[1] + elfs[2]);
}
